ocean_provider/routes/consume.py

```diff
@@ -284,6 +287,86 @@ def download():
         )
 
 
+@services.route("/assetUrls", methods=["GET"])
+@validate(AssetUrlsRequest)
+def asset_urls():
+    """Lists files from an asset
+    ---
+    tags:
+      - services
+    consumes:
+      - application/json
+    parameters:
+      - name: documentId
+        in: query
+        description: The ID of the asset/document (the DID).
+        required: true
+        type: string
+      - name: signature
+        in: query
+        description: Signature of the documentId to verify that the consumer has rights to download the asset.
+      - name: serviceId
+        in: string
+        description: ServiceId for the asset access service.
+      - name: nonce
+        in: string
+        description: User nonce.
+      - name: publisherAddress
+        in: query
+        description: The publisher address.
+        required: true
+        type: string
+    responses:
+      200:
+        description: lists asset files
+      400:
+        description: One of the required attributes is missing.
+      503:
+        description: Service Unavailable
+    """
+    data = get_request_data(request)
+    did = data.get("documentId")
+    service_id = data.get("serviceId")
+    publisher_address = data.get("publisherAddress")
+
+    logger.info(f"asset urls endpoint called. {data}")
+    try:
+        asset = get_asset_from_metadatastore(get_metadata_url(), did)
+        service = asset.get_service_by_id(service_id)
+        assert service.type == "access"
+
+        dt_token = get_web3().eth.contract(
+            abi=ERC20Template.abi, address=service.datatoken_address
+        )
+
+        if not dt_token.caller.isMinter(publisher_address):
```

Comment:
this is not right.
we need to check for nft owner.

Reply:
gotcha, will do.